compiler/rustc_privacy/src/lib.rs

```diff
@@ -292,6 +292,7 @@ where
             | ty::Bound(..)
             | ty::Error(_)
             | ty::GeneratorWitness(..)
+            | ty::Alias(ty::Weak, ..)
```

Comment:
this differs from projections in that the following errors for projections
```rust
trait Foo {
    type Assoc;
}

pub fn foo<T: Foo>() -> <T as Foo>::Assoc {
    todo!()
}
```
similarly i would expect the following to error:
```rust
type Foo = (impl Sized, u8);
pub fn foo() -> Foo {
    todo!()
}
```

Reply:
Oh I like that, so many surprisingly positive effects from making these type aliases their own type